workshop/Lab3/README.md

```diff
@@ -0,0 +1,291 @@
+Every Object that we create, we should provide the `apiVersion` you are using to create the object, `kind` of the object you are creating and the `metadata` about the object such as a `name`, set of `labels` and optionally `namespace` that this object should belong. 
+
+Consider the following deployment configuration for guestbook application
+```
+    apiVersion: apps/v1
+	kind: Deployment
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  replicas: 3
+	  selector:
+	    matchLabels:
+	      app: guestbook
+	  template:
+	    metadata:
+	      labels:
+	        app: guestbook
+	    spec:
+	      containers:
+	      - name: guestbook
+	        image: ibmcom/guestbook:v1
+	        ports:
+	        - name: http-server
+	          containerPort: 3000
+```
+The above configuration file create a deployment object named 'guestbook' with a pod containing a single container running the image 'ibmcom/guestbook:v1'.  Also the configuration specifies replicas set to 3 and kubernetes tries to make sure that at least three active pods running at any time.
+	
+- Create guestbook deployment 
+
+   ``` $kubectl create -f guestbook-deployment.yaml```
+
+- List the pod with label app=guestbook 
+
+   ``` $kubectl get pods -l app=guestbook```
+   
+You can simply change the deployment, alter replicas using the following command. Kubernetes then sees the desired the you want and tries to reconcile the actual current state to the desired state. 
+
+When you change the number of replicas in the configuration, Kubernetes will try to add, or remove, pods from the system to match your request. To can make these modifications by using the following command:
+
+```	
+	 $kubectl edit deployment guestbook 
+```
+
+You can also edit the deployment file locally to make changes. You should use the following command to make the change effective when you edit the deployment locally.
+
+   ``` $kubectl apply -f guestbook-deployment.yaml```
+
+We now define a service object to expose the deployment to external clients.
+```
+	apiVersion: v1
+	kind: Service
+	metadata:
+	  name: guestbook
+	  labels:
+	    app: guestbook
+	spec:
+	  ports:
+	  - port: 3000
+	    targetPort: http-server
+	  selector:
+	    app: guestbook
+	  type: LoadBalancer
+``` 
+The above configuration creates a Service resource named guestbook. A Service can be used
+to create a network path for incoming traffic to your running application.
+In this case, we are setting up a route from port 3000 on the cluster to the
+"http-server" port on our app, which is port 3000 per the Deployment container spec.
+Note that we create a "LoadBalancer" type of Service which will route all incoming
+traffic to each of the instances of our application - e.g. in a round-robin algorithm.
+
+- Let us now create guestbook service.
```

Comment:
s/create guestbook/create the guestbook/

Reply:
done